Signed Comparision warning with gcc 9.1

Upgraded to gcc 9.1.0 and received the following error:

```
[  0%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/encryption/encryption_openssl.cc.o
/home/seth/git/TileDB/tiledb/sm/encryption/encryption_openssl.cc: In static member function ‘static tiledb::sm::Status tiledb::sm::OpenSSL::encrypt_aes256gcm(tiledb::sm::ConstBuffer*, tiledb::sm::ConstBuffer*, tiledb::sm::ConstBuffer*, tiledb::sm::Buffer*, tiledb::sm::PreallocatedBuffer*, tiledb::sm::PreallocatedBuffer*)’:
/home/seth/git/TileDB/tiledb/sm/encryption/encryption_openssl.cc:70:21: error: comparison of integer expressions of different signedness: ‘uint64_t’ {aka ‘long unsigned int’} and ‘int’ [-Werror=sign-compare]
   70 |   if (input->size() > std::numeric_limits<int>::max())
      |       ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/seth/git/TileDB/tiledb/sm/encryption/encryption_openssl.cc: In static member function ‘static tiledb::sm::Status tiledb::sm::OpenSSL::decrypt_aes256gcm(tiledb::sm::ConstBuffer*, tiledb::sm::ConstBuffer*, tiledb::sm::ConstBuffer*, tiledb::sm::ConstBuffer*, tiledb::sm::Buffer*)’:
/home/seth/git/TileDB/tiledb/sm/encryption/encryption_openssl.cc:163:21: error: comparison of integer expressions of different signedness: ‘uint64_t’ {aka ‘long unsigned int’} and ‘int’ [-Werror=sign-compare]
  163 |   if (input->size() > std::numeric_limits<int>::max())
      |       ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors

```

--- a/tiledb/sm/encryption/encryption_openssl.cc
+++ b/tiledb/sm/encryption/encryption_openssl.cc
@@ -67,7 +67,7 @@ Status OpenSSL::encrypt_aes256gcm(
     PreallocatedBuffer* output_iv,
     PreallocatedBuffer* output_tag) {
   // Check input size for int datatype used by OpenSSL.
-  if (input->size() > std::numeric_limits<int>::max())
+  if (input->size() > static_cast<uint64_t>(std::numeric_limits<int>::max()))
     return LOG_STATUS(Status::EncryptionError(
         "OpenSSL error; cannot encrypt: input too large"));
 
@@ -160,7 +160,7 @@ Status OpenSSL::decrypt_aes256gcm(
     ConstBuffer* input,
     Buffer* output) {
   // Check input size for int datatype used by OpenSSL.
-  if (input->size() > std::numeric_limits<int>::max())
+  if (input->size() > static_cast<uint64_t>(std::numeric_limits<int>::max()))
     return LOG_STATUS(Status::EncryptionError(
         "OpenSSL error; cannot decrypt: input too large"));
 

